### PR TITLE
github/template: add pull request template

### PR DESCRIPTION
**Pull request by shashankram**
Adds a pull request template to make pull request descriptions
more consistent and to help understand their scope.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+- [ ] Certificate Management
+- [ ] Networking
+- [ ] Metrics
+- [ ] Security
+- [ ] Tests / CI System
+- [ ] Other
+
+Please answer the following questions with yes/no.
+
+- Is there code borrowed from external sources (provide references if yes):
```
Comment (lachie83) [inline .github/pull_request_template.md:20]:
```suggestion
- Does this change contain code from or inspired by another project? If so, did you notify the maintainers give appropriate attribution?
```

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+- [ ] Certificate Management
+- [ ] Networking
+- [ ] Metrics
+- [ ] Security
+- [ ] Tests / CI System
+- [ ] Other
+
+Please answer the following questions with yes/no.
+
+- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?
```
Comment (lachie83) [inline .github/pull_request_template.md:20]: Looks like we lost the yes/no

Comment (shashankram) [reply .github/pull_request_template.md:20]: Didn't follow, what do you suggest we add?

Comment (shashankram) [reply .github/pull_request_template.md:20]: The `yes/no` checkbox was dropped intentionally to make users answer the question explicitly. Having a checkbox can lead to accidental marking of checkbox which could lead to serious problems, so making users answer with `yes/no` seems like a good choice.

**Pull request merged**